lib/util/xml_utils.js

```diff
@@ -402,3 +420,23 @@ shaka.util.XmlUtils = class {
     }
   }
 };
+
+/**
+ * Promote a string to TrustedHTML. This function is security-sensitive and
+ * should only be used with security approval where the string is guaranteed not
+ * to cause an XSS vulnerability.
```

Comment:
So in the current use-case, just promoting to trusted is acceptable because we are performing that namespace check later in the function?

Reply:
Yes. The DOMParser.parseFromString API is guarded by Trusted Types because it can produce HTML DOM nodes (especially when parsing as text/html), which may then (intentionally or not) be inserted into a document, causing script execution. The output when parsing text/xml is usually safe, as the elements are typically from non-HTML namespaces and will be treated as inert by the browser. This PR closes the gap for when the parsed XML contains elements explicitly from the HTML/SVG namespaces, making the XML parser safe to use. The temporary conversion to TrustedHTML is safe even though the string is potentially user-controlled, as the TrustedHTML value is only used to produce (but not use) an XmlDocument, and that document is only returned for further use after verifying that it only consists of inert elements.